### DISCUSSION TITLE
Reprocessing errored messages in one channel from another channel

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5846] [Q&A]
I tried ChatGPT for some code to do this. It brought back the below:

```
var channelName = 'Quest Orders Test'; // Replace with the name of your Mirth channel 

// Connect to Mirth Connect server
var server = Packages.com.mirth.connect.server.controllers.ControllerFactory.getFactory().createController(); 

// Get the channel object by name
var channel = server.getChannel(channelName); 

// Retrieve the errored messages for the channel
var erroredMessages = channel.getErroredMessages();

```

Mirth throws an error with createController:

```
Cannot find function createController in object com.mirth.connect.server.controllers.DefaultControllerFactory@29a0fdc2.
```

Is there a better way to connect to another channel to get the errored messages? I want to resubmit them within the same channel they errored in.

**Comment by rutledj** [comment]
Wording the question differently, it also returned some code to use the channelUtil.getErrorMessages() function but that doesn't show up in the channelUtil class as a valid function.

**Reply by tonygermano** [reply]
I think the takeaway here is that chatGPT sucks at writing mirth code and invents things that don't exist.

**Comment by tonygermano** [comment]
I'm not sure I understand your goal. If you want messages with an error status to automatically retry, then I think you want to use a destination queue instead?
